src/main/java/uk/ac/ebi/embl/gff3tools/validation/builtin/GeneFeatureValidation.java

```diff
@@ -39,131 +42,121 @@ public class GeneFeatureValidation extends Validation {
-                                geneName, GFF3Attributes.LOCUS_TAG, existingLocus, locusTag));
+                                geneName, GFF3Attributes.PSEUDOGENE, existingPseudo, pseudoGeneName));
             }
+            geneToPseudoGene.put(geneName, pseudoGeneName);
         }
-        geneToLocusTag.put(geneName, locusTag);
-
-        String pseudoGeneName = feature.getAttributeByName(GFF3Attributes.PSEUDOGENE);
-        String existingPseudo = geneToPseudoGene.get(geneName);
-        if (existingPseudo != null && !Objects.equals(existingPseudo, pseudoGeneName)) {
-            throw new ValidationException(
-                    line,
-                    GENE_ASSOCIATION_VALIDATION.formatted(
-                            geneName, GFF3Attributes.PSEUDOGENE, existingPseudo, pseudoGeneName));
-        }
-        geneToPseudoGene.put(geneName, pseudoGeneName);
     }
 
-    @ValidationMethod(rule = "GENE_LOCUS_TAG_ASSOCIATION", type = ValidationType.FEATURE)
-    public void validateGeneLocusTagAssociation(GFF3Feature feature, int line) throws ValidationException {
-        String featureName = feature.getName();
-        Optional<String> soIdOpt = ontologyClient.findTermByNameOrSynonym(featureName);
-
-        if (soIdOpt.isEmpty()) {
-            return;
-        }
+    @ValidationMethod(rule = "GENE_LOCUS_TAG_ASSOCIATION", type = ValidationType.ANNOTATION)
+    public void validateGeneLocusTagAssociation(GFF3Annotation gff3Annotation, int line) throws ValidationException {
+        Map<String, GFF3Feature> locusTagToGeneFeature = new HashMap<>();
+        for (GFF3Feature feature : gff3Annotation.getFeatures()) {
+            Optional<String> soIdOpt = ontologyClient.findTermByNameOrSynonym(feature.getName());
 
-        String soId = soIdOpt.get();
+            if (soIdOpt.isEmpty()) continue;
 
-        // Check if feature is a gene or pseudogene type
-        boolean isGene = soId.equals(OntologyTerm.GENE.ID)
-                || soId.equals(OntologyTerm.PSEUDOGENE.ID)
-                || soId.equals(OntologyTerm.UNITARY_PSEUDOGENE.ID)
-                || ontologyClient.isSelfOrDescendantOf(soId, OntologyTerm.PSEUDOGENE.ID)
-                || ontologyClient.isSelfOrDescendantOf(soId, OntologyTerm.UNITARY_PSEUDOGENE.ID);
+            String soId = soIdOpt.get();
 
-        if (!isGene) {
-            return;
-        }
-        Map<String, GFF3Feature> locusTagToGeneFeature =
-                annotationLocusTagToGeneFeature.computeIfAbsent(feature.accession(), k -> new HashMap<>());
+            // Check if feature is a gene or pseudogene type
+            boolean isGene = soId.equals(OntologyTerm.GENE.ID)
+                    || soId.equals(OntologyTerm.PSEUDOGENE.ID)
+                    || soId.equals(OntologyTerm.UNITARY_PSEUDOGENE.ID)
+                    || ontologyClient.isSelfOrDescendantOf(soId, OntologyTerm.PSEUDOGENE.ID)
+                    || ontologyClient.isSelfOrDescendantOf(soId, OntologyTerm.UNITARY_PSEUDOGENE.ID);
```

Comment:
Same as above, if `ontologyClient.isSelfOrDescendantOf `is enough we can ignore `soId.equals(`